static/app/utils/useNow.tsx

```diff
@@ -0,0 +1,22 @@
+import {useRef} from 'react';
+import identity from 'lodash/identity';
+
+/**
+ * Hook which returns a Date object which is only instantiated the first time
+ * the hook is called.
+ *
+ * A modifier callback may be provided to have the date configured after it is
+ * first initialized.
+ */
+export function useNow(modifier?: (now: Date) => Date) {
```

Comment:
I think this is an unnecessary abstraction for a single liner.

Reply:
There is an advantage with this implementation that it doesn't create a new Date on every render, which I think makes the abstraction ever so slightly worth it.

Reply:
imho useMemo achieves the same thing right?

Reply:
>You should only rely on useMemo as a performance optimization. If your code doesn’t work without it, find the underlying problem and fix it first. Then you may add useMemo to improve performance.

useMemo wouldn't make sense here since it might re-calculate the value (and create a new date) without notice.